### PR TITLE
[SPARK-46876]csv line containing delimiter can't be treated as empty line

### PR DESCRIPTION
**Pull request by doki23**
### What changes were proposed in this pull request?

Don't treat csv line which only containing delimiters as an empty line in case of being filtered.

### Why are the changes needed?

[SPARK-46876](https://issues.apache.org/jira/projects/SPARK/issues/SPARK-46876)

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Not yet, WIP

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala [hunk 1]
```diff
@@ -25,13 +25,18 @@ object CSVExprUtils {
    * This is currently being used in CSV reading path and CSV schema inference.
    */
   def filterCommentAndEmpty(iter: Iterator[String], options: CSVOptions): Iterator[String] = {
+    @inline
+    def nonEmptyLine(line: String) : Boolean = {
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVExprUtils.scala:29]: should also fix `org.apache.spark.sql.execution.datasources.csv.CSVUtils.filterCommentAndEmpty`

**Comment by martinitus** [comment]
Some unit-test that will prevent regression would be nice :)

Thanks for fixing this!

**Comment by doki23** [comment]
It's ready for review.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVUtils.scala [hunk 2]
```diff
@@ -35,6 +34,7 @@ object CSVUtils {
     import lines.sqlContext.implicits._
     val aliased = lines.toDF("value")
     val nonEmptyLines = aliased.filter(length(trim($"value")) > 0)
+//      aliased.filter(length(trim($"value")) > 0 or contains($"value", lit(options.delimiter)))
```
Comment (martinitus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVUtils.scala]: remove dead code?

**Pull request closed without merging**